src/components/eventsPage/eventsPage.tsx

```diff
@@ -1,45 +1,94 @@
+import { getTranslations } from "next-intl/server";
+
 import EventPosting from "src/components/eventPosting/EventPosting";
+import Text from "src/components/text/Text";
 import { IEventPosting } from "studio/lib/interfaces/eventPosting";
 
-import EventsHeader from "./eventsHeader";
 import styles from "./eventsPage.module.css";
 
+const sortByDateAsc = (a: IEventPosting, b: IEventPosting) => {
+  const dateA = a.date ? new Date(a.date).getTime() : Infinity;
+  const dateB = b.date ? new Date(b.date).getTime() : Infinity;
+  return dateA - dateB;
+};
+
 export default async function EventsPage({
   params,
   eventPostings,
 }: {
   params: { locale: string };
   eventPostings: { eventPostings: IEventPosting[] };
 }) {
+  const t = await getTranslations("event_section");
+  const allEventsId = "all-events-id";
+  const futureEventsId = "future-events-id";
+  const pastEventsId = "past-events-id";
   const today = new Date();
   const yesterday = new Date(today.setDate(today.getDate() - 1));
 
-  const filteredEventPostings = eventPostings.eventPostings.filter((event) => {
+  const futureEventPostings = eventPostings.eventPostings.filter((event) => {
     const eventDate = event.date ? new Date(event.date) : null;
     if (!eventDate) return true;
     return eventDate >= yesterday;
   });
 
-  const limitedEventPostings = filteredEventPostings.sort((a, b) => {
-    const dateA = a.date ? new Date(a.date).getTime() : Infinity;
-    const dateB = b.date ? new Date(b.date).getTime() : Infinity;
-    return dateA - dateB;
+  const pastEventPostings = eventPostings.eventPostings.filter((event) => {
+    const eventDate = event.date ? new Date(event.date) : null;
+    if (!eventDate) return true;
+    return eventDate < yesterday;
   });
 
+  const sortedFutureEventPostings = futureEventPostings.sort(sortByDateAsc);
+  const sortedPastEventPostings = pastEventPostings
+    .sort(sortByDateAsc)
+    .reverse();
+
   return (
-    <div className={styles.eventSection}>
-      <EventsHeader />
-
-      <div className={styles.wrapper}>
-        {limitedEventPostings.map((event: IEventPosting) => (
-          <EventPosting
-            eventPosting={event}
-            key={event._key}
-            showLocations={true}
-            language={params.locale}
-          />
-        ))}
-      </div>
-    </div>
+    <>
+      <section className={styles.eventSection} aria-labelledby={allEventsId}>
+        <Text type="h1" id={allEventsId}>
+          {t("all_events")}
+        </Text>
```

Comment:
Reading this there is something odd about "All events" and further down "Previous events". It isn't intuitive to me that the events on the top is "upcoming". Especially when past and future events have the exact same styling and is visually similarly prioritised. That being said, I think having them shown adds more value than it confuses, and I know changing style / thinking about this properly will probably mean that it won't happen at first.

Reply:
Agree but I didn't want to change anything on design. 
- There is a visually hidden heading for "Coming events" that could be made non-hidden. 
- Past events could probably have a smaller card for the events than future.
- Past events should probably be paginated.
- My OCD suffers from the difference in sorting between the lists one ASC one DESC. But to me the past list made no sense being ASC, and it actually makes some sense to have the future events ASC. Maybe a third larger size of card could be used for the next event making it possible to sort the future events DESC as well but still getting focus to the next upcoming event.

All this could be addressed when need be.